src/CoreWCF.Http/tests/ContractBehaviorTests.cs

```diff
@@ -0,0 +1,311 @@
+            try
+            {
+                var theBehavior = new MyMultiFacetedBehaviorAttribute();
+                cf.Endpoint.Contract.ContractBehaviors.Add(theBehavior);
+                cf.Open();
+                string expected = "IContractBehavior:ClientContract.MyMultiFacetedBehaviorAttribute;";
+                Assert.Equal(expected, BehaviorInvokedVerifier.ValidateClientInvokedBehavior(cf.Endpoint));
+                IContractBehaviorBasic_ByHand clientProxy = cf.CreateChannel();
+                string HelloStr = "ByHandImplementsOther";
+                string returnStr = clientProxy.StringMethod(HelloStr);
+                Assert.Equal(HelloStr, returnStr);
+            }
+            catch
+            {
+                throw;
+            }
+            finally
+            {
+                if (cf != null && cf.State == System.ServiceModel.CommunicationState.Opened)
+                    cf.Close();
+            }
+        }
+
+        internal class Startup
+        {
+            public static string _method = "";
+            public void ConfigureServices(IServiceCollection services)
+            {
+                services.AddServiceModelServices();
+            }
+
+            public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+            {
+                app.UseServiceModel(builder =>
+                {
+                    switch(_method)
+                    {                     
+                        case "ByHand":
+                        case "ByHand_UsingHiddenProperty":
+                            builder.AddService<ContractBehaviorBasic_ByHand_Service>();
+                            builder.AddServiceEndpoint<ContractBehaviorBasic_ByHand_Service, ServiceContract.IContractBehaviorBasic_ByHand>(new BasicHttpBinding(), "/BasicWcfService/ContractBehaviorService.svc");
+                            builder.ConfigureServiceHostBase<ContractBehaviorBasic_ByHand_Service>(serviceHost =>
+                            {
+                                var cb = new ServiceContract.CustomContractBehaviorAttribute();
+                                serviceHost.Description.Endpoints[0].Contract.ContractBehaviors.Add(cb);
+                            });
+                            break;
+                        case "ByHandImplementsOther":
+                            builder.AddService<ContractBehaviorBasic_ByHand_Service>();
+                            builder.AddServiceEndpoint<ContractBehaviorBasic_ByHand_Service, ServiceContract.IContractBehaviorBasic_ByHand>(new BasicHttpBinding(), "/BasicWcfService/ContractBehaviorService.svc");
+                            builder.ConfigureServiceHostBase<ContractBehaviorBasic_ByHand_Service>(serviceHost =>
+                            {
+                                var cb = new ServiceContract.MyMultiFacetedBehaviorAttribute();
+                                serviceHost.Description.Endpoints[0].Contract.ContractBehaviors.Add(cb);
```

Comment:
@mconnew  now I am able to add custom ContractBehaviors this way ("by hand"), but test failed when validating the behavior implementation, the code [here](https://github.com/CoreWCF/CoreWCF/blob/8e24169c70fcf83e4be6a6e9602fd5d3b66dfa93/src/CoreWCF.Http/tests/ServiceContract/IContractBehavior.cs#L63) didn't execute.  

The other test variation which use the custom behavior as an attribute decorating the contract interface works, the ApplyDispatchBehavior(...) implementation is called with test flag being set as expected.

Reply:
The affected "byhand" test scenario validation has been skipped, tracking issue: #193.